### DISCUSSION TITLE
Major Update #2: LayerDiffuse Transparent Image Editing

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/stable-diffusion-webui-forge#854] [Announcements]
Thanks to Gradio 4, new canvas, and some backend works, now transparent image editing finally landed.

# First Read This If You Have Not

Read this first:

https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/853

# Layer Diffuse

Install extension from here:

https://github.com/lllyasviel/sd-forge-layerdiffuse

# Example 1: inpaint

Given this image (Note that this image is transparent):

![causal_cut](https://github.com/user-attachments/assets/4239120b-5816-404b-8ca8-2e5e7fb4bb5f)

Your goal is to improve the boundary to make it usable

You can draw like this (this is my draw history for your reference):

![2](https://github.com/user-attachments/assets/53984c09-c1c8-4a41-b6e0-d5d2db9b8a9c)

And you can use img2img to achieve the editing now:

![aa](https://github.com/user-attachments/assets/b104d411-7c5c-4576-95db-21d027512780)
![3](https://github.com/user-attachments/assets/7ba5a6c2-4c1c-458b-bcc1-a87116c888b8)

# Example 2: sketch + img2img

Given this image (Note that this image is transparent):

![00070-286627024](https://github.com/user-attachments/assets/f920038f-56e5-4544-b216-04c0f4ccc823)

Your goal is to add some fire glowing to make it burn

You can draw like this (this is my draw history for your reference):

![1](https://github.com/user-attachments/assets/1d70e24d-d157-4c62-b608-c4783ecf77e0)

And you can use img2img to achieve the editing now:

![aa](https://github.com/user-attachments/assets/b104d411-7c5c-4576-95db-21d027512780)
![bb](https://github.com/user-attachments/assets/92b27980-ab6a-4a50-9834-e4ee92127bb0)

**Comment by Gushousekai195** [comment]
When does IPAdapter get updated?

**Comment by Gushousekai195** [comment]
I would like to be able to style transfer with IPAdapter in Forge, please.

**Comment by mel870** [comment]
Thanks, this is a great update!

**Comment by DiegoRRR** [comment]
AWESOME!!! I must admit I had lost hope. Thank you so much for fixing img2img and for all your hard work on this.

**Comment by zhaol4v** [comment]
amazing work, this is huge

**Comment by Susanoo1337** [comment]
What are the exact constraints for image size when using layerdiffuse? Obviously 1024x1024, 1024x704, 1408x640 works... I've been trying to find more working dimensions but it's tedious. Every other HxW gives a Runtime Error about tensor dimension mismatch.

**Comment by elismasilva** [comment]
i was trying discover what was the commit hash where you introduce img2img i didnt found, i mean on layerdiffuse repository

**Comment by gigend** [comment]
it's support for flux bnb nf4?

**Comment by wktra** [comment]
more junk. doesn't work. Makes random spots transparent when used in img2img

**Reply by rungvang** [reply]
Just use the birefnet for background removal in "Space tab". For drawing + inpaint, use Krita.

**Comment by VotreLiberte53** [comment]
Hi, am I correct in understanding that LayerDiffuse does not support the use of Lora?

**Reply by Wontfallo** [reply]
ever get this working?

**Comment by writecai** [comment]
LayerDiffuse forge support flux how long?
